subworkflows/local/utils_nfcore_rnaseq_pipeline/main.nf

```diff
@@ -254,15 +254,12 @@ def validateInputParameters() {
 }
 
 //
-// Get attribute from genome config file e.g. fasta
+// Get key from genome config file e.g. fasta, and initialize a params with the given attribute
 //
-def getGenomeAttribute(attribute) {
-    if (params.genomes && params.genome && params.genomes.containsKey(params.genome)) {
-        if (params.genomes[ params.genome ].containsKey(attribute)) {
-            return params.genomes[ params.genome ][ attribute ]
-        }
-    }
-    return null
+def getGenomeAttribute(key, attribute) {
+    if ((params.genomes && params.genome && params.genomes.containsKey(params.genome) && params.genomes[ params.genome ].containsKey(key) )) {
+        params."$attribute" = params.genomes[ params.genome ][ key ]
+    } else params."$attribute" = null
```

Comment:
other idea, dynamic params.
works well, but not super readable.
and we'd need to add the params for which the names aren't the igenomes key in `validationSchemaIgnoreParams`

Reply:
yeah, this is not working properly...